doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_clockman.rst

```diff
@@ -0,0 +1,111 @@
+.. _ug_nrf54h20_architecture_clockman:
+
+nRF54H20 Clock Management
+#########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The nRF54H20 SoC consists of multiple asynchronous clock domains and requires clock sources for correct operation.
+Each of the clock sources needs proper management:
+
+* To start and stop the clock at the appropriate time.
+* To select the proper clock source.
+* To calibrate or synchronize the clock to a more accurate source.
+
+Some of the clock management operations are performed solely by the hardware circuits, but others require software intervention.
+Most of the clocks can be locked to other, more accurate ones to improve their accuracy.
+
+Clock Domains
+*************
+
+The nRF54H20 SoC consists of the following clock domains:
+
+* LFCLK
+* HFXO
+* FLL16M
+* HSFLL Application
+* HSFLL Radio (if Radio domain is present)
+* HSFLL Secure
+* HSFLL Global
+
+Each clock domain is enabled independently of others and it is automatically enabled when at least one sink is active.
+The firmware can choose to keep each clock domains enabled even when all its sinks are inactive.
+
+Clock domain source selection differs between clock domains:
+
+* The LFCLK source is selected by the System Controller Firmware.
+* The HFXO source is selected by the System Controller Firmware.
+* The FLL16M source is selected by local domains or System Controller Firmware (SCFW) in all scenarios, except one:
+  When the firmware selects open-loop, but any of the HSFLL sources is in closed-loop, the FLL16M is automatically switched by the hardware to closed-loop.
+
+* HSFLLs sources are selected by firmware (depending on an HSFLL instance: local domain or System Controller).
```

Comment:
I would rephrased this part to somethinkg like

"
Clock domain sources are selected by clock_control drivers based on the clock parameters' requirements reported by the software modules using the clock_control functions. The requirements may cover clock frequency, accuracy, or precision. Some clock domains are configured by clock_control drivers using LRCCONF peripherals, others are configured with assistance of the System Controller Firmware.
"

I'm not sure if more details is required in the public documentation.